src/bokeh/models/dom.py

```diff
@@ -170,12 +173,34 @@ def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
 class ValueRef(Placeholder):
+    """ Allows to reference a value in a column of a data source.
+    """
 
     # explicit __init__ to support Init signatures
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
-    field = Required(String)
+    field = Required(String, help="""
+    The name of the field to reference, which is equivalent to using ``"@{field}``.
+    """)
+
+    format = Nullable(String, default=None, help="""
+    Optional format string, which is equivalent to using ``"@{field}{format}"``.
+    """)
+
+    formatter = Either(
+        Enum(BuiltinFormatter),
+        Instance(".models.callbacks.CustomJS"),
+        Instance(".models.tools.CustomJSHover"), default="raw", help="""
+    Either a named value formatter or an instance of ``CustomJS`` or ``CustomJSHover``.
+
+    .. note::
+        Custom JS formatters can return a value of any type, not necessarily a string.
+        If a non-string value is returned then, if it's an instance of DOM ``Node``
+        (in particular it can be a DOM ``Document`` or a ``DocumentFragment``), then
```

Comment:
Does this add a reference to the class in the documentation? It can be somewhat challenging to know what `Node`, `Document`, and `DocumentFragment` are.

Reply:
Those are native DOM classes, thus this would have to point to MDN or W3C, which isn't a bad idea.